core/domain/story_domain.py

```diff
@@ -565,16 +638,23 @@ def get_ordered_nodes(self):
         Returns:
             list(StoryNode). The ordered list of nodes.
         """
+        # Ruling out the possibility of None for mypy type checking.
+        assert self.initial_node_id is not None
         initial_index = self.get_node_index(self.initial_node_id)
+        # Ruling out the possibility of None for mypy type checking.
+        assert initial_index is not None
```

Comment:
As we need to use `assert` a lot here, should we introduce a version of `get_node_index` that would automatically do this check?

Reply:
Added a new version of get_node_index which is `strict_get_node_index`. This version only returns int and None it raises the valuerror.